diff --git a/dist/serviceGenerator.js b/dist/serviceGenerator.js
index 001071c0a33034c7ca51603cfbc7801488172dfa..90a50d71139be1fb8644ed125d3bed22e5ab0721 100644
--- a/dist/serviceGenerator.js
+++ b/dist/serviceGenerator.js
@@ -66,21 +66,16 @@ const getType = (schemaObject, namespace = '') => {
     ];
     const dateEnum = ['Date', 'date', 'dateTime', 'date-time', 'datetime'];
     const stringEnum = ['string', 'email', 'password', 'url', 'byte', 'binary'];
-    if (numberEnum.includes(schemaObject.format)) {
-        type = 'number';
-    }
     if (schemaObject.enum) {
         type = 'enum';
-    }
-    if (numberEnum.includes(type)) {
+    } else if (numberEnum.includes(type)) {
         return 'number';
-    }
-    if (dateEnum.includes(type)) {
+    } else if (dateEnum.includes(type)) {
         return 'Date';
-    }
-    if (stringEnum.includes(type)) {
+    } else if (stringEnum.includes(type)) {
         return 'string';
     }
+
     if (type === 'boolean') {
         return 'boolean';
     }
@@ -99,7 +94,15 @@ const getType = (schemaObject, namespace = '') => {
     }
     if (type === 'enum') {
         return Array.isArray(schemaObject.enum)
-            ? Array.from(new Set(schemaObject.enum.map((v) => typeof v === 'string' ? `"${v.replace(/"/g, '"')}"` : getType(v)))).join(' | ')
+            ? Array.from(new Set(schemaObject.enum.map((v) => {
+                if(numberEnum.includes(schemaObject.format)) {
+                    return v.replace(/"/g, '"');
+                } else if(typeof v === 'string') {
+                    return `"${v.replace(/"/g, '"')}"`;
+                } else {
+                    return getType(v);
+                }
+            }))).join(' | ')
             : 'string';
     }
     if (schemaObject.oneOf && schemaObject.oneOf.length) {
@@ -108,7 +111,13 @@ const getType = (schemaObject, namespace = '') => {
     if (schemaObject.allOf && schemaObject.allOf.length) {
         return `(${schemaObject.allOf.map((item) => getType(item, namespace)).join(' & ')})`;
     }
-    if (schemaObject.type === 'object' || schemaObject.properties) {
+    if (schemaObject.type === 'object' || schemaObject.properties || schemaObject.additionalProperties ) {
+        if (!schemaObject.properties && !schemaObject.additionalProperties) {
+            return 'any';
+        }
+        if (schemaObject.additionalProperties) {
+            return `Record<string, ${getType(schemaObject.additionalProperties, namespace)}>`;
+        }
         if (!Object.keys(schemaObject.properties || {}).length) {
             return 'Record<string, any>';
         }
@@ -259,7 +268,7 @@ class ServiceGenerator {
             // functionName tag 级别防重
             const tmpFunctionRD = {};
             const genParams = this.apiData[tag]
-                .filter((api) => 
+                .filter((api) =>
             // 暂不支持变量
             !api.path.includes('${'))
                 .map((api) => {
@@ -273,8 +282,14 @@ class ServiceGenerator {
                     file = this.concatOrNull(file, newfile);
                     //const file = this.getFileTP(newApi.requestBody);
                     let formData = false;
-                    if ((body && (body.mediaType || '').includes('form')) || file) {
+                    let binaryFile = false;
+                    if (file) {
+                        formData = true;
+                    } else if (body && (body.mediaType || '').includes('form')) {
+                        formData = true;
+                    } else if (body && (body?.propertiesList || []).filter((v) => v.schema.format === 'binary').length > 0) {
                         formData = true;
+                        binaryFile = true;
                     }
                     let functionName = this.config.hook && this.config.hook.customFunctionName
                         ? this.config.hook.customFunctionName(newApi)
@@ -340,12 +355,12 @@ class ServiceGenerator {
                         // prefix 变量
                         return `$\{${prefix}}${formattedPath}`;
                     };
-                    return Object.assign(Object.assign({}, newApi), { functionName, path: getPrefixPath(), pathInComment: formattedPath.replace(/\*/g, '&#42;'), hasPathVariables: formattedPath.includes('{'), hasApiPrefix: !!this.config.apiPrefix, method: newApi.method, 
-                        // 如果 functionName 和 summary 相同，则不显示 summary
-                        desc: functionName === newApi.summary
-                            ? newApi.description
-                            : [newApi.summary, newApi.description].filter((s) => s).join(' '), hasHeader: !!(params && params.header) || !!(body && body.mediaType), params: finalParams, hasParams: Boolean(Object.keys(finalParams || {}).length), body,
-                        file, hasFormData: formData, response });
+                    return Object.assign(Object.assign({}, newApi), { functionName, path: getPrefixPath(), pathInComment: formattedPath.replace(/\*/g, '&#42;'), hasPathVariables: formattedPath.includes('{'), hasApiPrefix: !!this.config.apiPrefix, method: newApi.method,
+                    // 如果 functionName 和 summary 相同，则不显示 summary
+                    desc: functionName === newApi.summary
+                        ? newApi.description
+                        : [newApi.summary, newApi.description].filter((s) => s).join(' '), hasHeader: !!(params && params.header) || !!(body && body.mediaType), params: finalParams, hasParams: Boolean(Object.keys(finalParams || {}).length), body,
+                    file, hasFormData: formData, hasBinaryFile: binaryFile, response })
                 }
                 catch (error) {
                     // eslint-disable-next-line no-console
@@ -529,7 +544,7 @@ class ServiceGenerator {
         return schemaObject.properties
             ? Object.keys(schemaObject.properties).map((propName) => {
                 const schema = (schemaObject.properties && schemaObject.properties[propName]) || DEFAULT_SCHEMA;
-                return Object.assign(Object.assign({}, schema), { name: propName, type: getType(schema), desc: [schema.title, schema.description].filter((s) => s).join(' '), 
+                return Object.assign(Object.assign({}, schema), { name: propName, type: getType(schema), desc: [schema.title, schema.description].filter((s) => s).join(' '),
                     // 如果没有 required 信息，默认全部是非必填
                     required: requiredPropKeys ? requiredPropKeys.some((key) => key === propName) : false });
             })
diff --git a/templates/serviceController.njk b/templates/serviceController.njk
index 3a5b78e040d1a535fba3fd07e4f37955bc6e91bc..3609cb8ba93a373383b375f83630200bcdfeb2be 100644
--- a/templates/serviceController.njk
+++ b/templates/serviceController.njk
@@ -61,7 +61,13 @@ export async function {{ api.functionName }}(
     {% if prop.schema.description -%}
     /** {{ prop.schema.description }} */
     {% endif -%}
-    {{ prop.key }}{{ "?" if not prop.schema.requird }}: {{ prop.schema.type }},
+    {{ prop.key }}{{ "?" if not prop.schema.required }}:
+      {%- if prop.schema.format !== 'binary' -%}
+      {{ prop.schema.type }}
+      {%- else -%}
+      File
+      {%- endif -%}
+      ,
     {%- endfor %}
   }
   {%- else -%}
@@ -100,12 +106,18 @@ files
   {% if api.body %}
   Object.keys(body).forEach(ele => {
     {% if genType === "ts" %}
-    const item = (body as any)[ele];
+    const item = (body as any)[ele]; // eslint-disable-line ts/no-explicit-any
     {% else %}
     const item = body[ele];
     {% endif %}
     if (item !== undefined && item !== null) {
-      formData.append(ele, typeof item === 'object' ? JSON.stringify(item) : item);
+      if (item instanceof File) {
+        formData.append(ele, item);
+      } else {
+        for (let k in item) {
+          formData.append(k, item[k]);
+        }
+      }
     }
   });
   {% endif %}
@@ -119,7 +131,9 @@ files
     method: '{{ api.method | upper }}',
     {%- if api.hasHeader %}
     headers: {
-      {%- if api.body.mediaType %}
+      {%- if api.hasFormData %}
+      'Content-Type': 'multipart/form-data',
+      {%- elif api.body.mediaType %}
       'Content-Type': '{{ api.body.mediaType | safe }}',
       {%- endif %}
     },