diff --git a/dist/serviceGenerator.js b/dist/serviceGenerator.js
index 001071c0a33034c7ca51603cfbc7801488172dfa..19f91fb980303274a6a6df5845ed2844eb0ebf6a 100644
--- a/dist/serviceGenerator.js
+++ b/dist/serviceGenerator.js
@@ -108,7 +108,13 @@ const getType = (schemaObject, namespace = '') => {
     if (schemaObject.allOf && schemaObject.allOf.length) {
         return `(${schemaObject.allOf.map((item) => getType(item, namespace)).join(' & ')})`;
     }
-    if (schemaObject.type === 'object' || schemaObject.properties) {
+    if (schemaObject.type === 'object' || schemaObject.properties || schemaObject.additionalProperties ) {
+        if (!schemaObject.properties && !schemaObject.additionalProperties) {
+            return 'any';
+        }
+        if (schemaObject.additionalProperties) {
+            return `Record<string, ${getType(schemaObject.additionalProperties, namespace)}>`;
+        }
         if (!Object.keys(schemaObject.properties || {}).length) {
             return 'Record<string, any>';
         }
@@ -259,7 +265,7 @@ class ServiceGenerator {
             // functionName tag 级别防重
             const tmpFunctionRD = {};
             const genParams = this.apiData[tag]
-                .filter((api) => 
+                .filter((api) =>
             // 暂不支持变量
             !api.path.includes('${'))
                 .map((api) => {
@@ -272,9 +278,15 @@ class ServiceGenerator {
                     const newfile = this.getFileTP(newApi.requestBody);
                     file = this.concatOrNull(file, newfile);
                     //const file = this.getFileTP(newApi.requestBody);
+                    let hasBinary = false;
                     let formData = false;
-                    if ((body && (body.mediaType || '').includes('form')) || file) {
+                    if (file) {
+                        formData = true;
+                    } else if (body && (body.mediaType || '').includes('form')) {
+                        formData = true;
+                    } else if (body && (body.propertiesList || []).filter((v) => v.schema.format === 'binary').length > 0) {
                         formData = true;
+                        hasBinary = true;
                     }
                     let functionName = this.config.hook && this.config.hook.customFunctionName
                         ? this.config.hook.customFunctionName(newApi)
@@ -340,12 +352,12 @@ class ServiceGenerator {
                         // prefix 变量
                         return `$\{${prefix}}${formattedPath}`;
                     };
-                    return Object.assign(Object.assign({}, newApi), { functionName, path: getPrefixPath(), pathInComment: formattedPath.replace(/\*/g, '&#42;'), hasPathVariables: formattedPath.includes('{'), hasApiPrefix: !!this.config.apiPrefix, method: newApi.method, 
+                    return Object.assign(Object.assign({}, newApi), { functionName, path: getPrefixPath(), pathInComment: formattedPath.replace(/\*/g, '&#42;'), hasPathVariables: formattedPath.includes('{'), hasApiPrefix: !!this.config.apiPrefix, method: newApi.method,
                         // 如果 functionName 和 summary 相同，则不显示 summary
                         desc: functionName === newApi.summary
                             ? newApi.description
                             : [newApi.summary, newApi.description].filter((s) => s).join(' '), hasHeader: !!(params && params.header) || !!(body && body.mediaType), params: finalParams, hasParams: Boolean(Object.keys(finalParams || {}).length), body,
-                        file, hasFormData: formData, response });
+                        file, hasFormData: formData, hasBinary, response });
                 }
                 catch (error) {
                     // eslint-disable-next-line no-console
@@ -529,7 +541,7 @@ class ServiceGenerator {
         return schemaObject.properties
             ? Object.keys(schemaObject.properties).map((propName) => {
                 const schema = (schemaObject.properties && schemaObject.properties[propName]) || DEFAULT_SCHEMA;
-                return Object.assign(Object.assign({}, schema), { name: propName, type: getType(schema), desc: [schema.title, schema.description].filter((s) => s).join(' '), 
+                return Object.assign(Object.assign({}, schema), { name: propName, type: getType(schema), desc: [schema.title, schema.description].filter((s) => s).join(' '),
                     // 如果没有 required 信息，默认全部是非必填
                     required: requiredPropKeys ? requiredPropKeys.some((key) => key === propName) : false });
             })
diff --git a/templates/serviceController.njk b/templates/serviceController.njk
index 3a5b78e040d1a535fba3fd07e4f37955bc6e91bc..fb8b66e3842f09d71916a2cf0dc6e49eef1f328b 100644
--- a/templates/serviceController.njk
+++ b/templates/serviceController.njk
@@ -61,7 +61,13 @@ export async function {{ api.functionName }}(
     {% if prop.schema.description -%}
     /** {{ prop.schema.description }} */
     {% endif -%}
-    {{ prop.key }}{{ "?" if not prop.schema.requird }}: {{ prop.schema.type }},
+    {{ prop.key }}{{ "?" if not prop.schema.required }}:
+      {%- if prop.schema.format !== 'binary' -%}
+      {{ prop.schema.type }}
+      {%- else -%}
+      File
+      {%- endif -%}
+      ,
     {%- endfor %}
   }
   {%- else -%}
@@ -105,7 +111,18 @@ files
     const item = body[ele];
     {% endif %}
     if (item !== undefined && item !== null) {
+      {%- if api.hasBinary -%}
+      if (typeof item === 'object') {
+        for (let k of item) {
+          const v = item[k];
+          formData.append(k, v);
+        }
+      } else {
+        formData.append(ele, item);
+      }
+      {%- else -%}
       formData.append(ele, typeof item === 'object' ? JSON.stringify(item) : item);
+      {%- endif -%}
     }
   });
   {% endif %}
@@ -119,7 +136,9 @@ files
     method: '{{ api.method | upper }}',
     {%- if api.hasHeader %}
     headers: {
-      {%- if api.body.mediaType %}
+      {%- if api.hasFormData %}
+      'Content-Type': 'multipart/form-data',
+      {%- elif api.body.mediaType %}
       'Content-Type': '{{ api.body.mediaType | safe }}',
       {%- endif %}
     },